serving/purgatorio/select-and-train-machine-learning-models/tools-and-libraries/tools-and-libraries.md

```diff
@@ -1,5 +1,321 @@
 ---
 title: Tools and Libraries
 author: clone95
-description: Learn to use existing libraries for Machine Learning and Deep Learning and out-of-the-box tools for productions systems.
----
\ No newline at end of file
+description: Learn to use existing libraries for Machine Learning and Deep Learning and out-of-the-box tools for production systems.
+---
+
+# Index
+- [How this guide is designed](#How-this-guide-is-designed)
+- [Solving the Learning Problem](#Solving-the-Learning-problem)
+- [Navigating the Ocean](#Navigating-the-Ocean)
+- [The Data Science Process Toolkit](#The-Data-Science-Process-Toolkit) 
+    + [Data Collection](#Data-Collection)
+    + [Data Preparation](#Data-Preparation)
+    + [Data Visualization](#Data-Visualization)
+    + [Machine Learning](#Machine-Learning)
+    + [Deep Learning](#Deep-Learning)
+    + [Hyperparameter Tuning](#Hyperparameter-Tuning)
+    + [Serving Models](#Serving-Models)
+    + [Miscellaneous](#Miscellaneous)
+
+- [Production Ready Tools](#Production-Ready-Tools)
+    + [Computer Vision](#Computer-vision)
+    + [Natural Language](#Natural-Language)
+    + [Time Series Forecasting](#Time-Series-Forecasting)
+
+- [Conclusions](#Conclusions)
+
+# How this guide is designed
+
+This guide is designed in the same fashion as the **Purgatorio** structure, so you find a subsection for each phase of the Data Science process.
+
+For _each section_, we list for you different libraries and the best resources to learn them.
+
+We'll give you some general tips to learn effectively and develop rock-solid foundations, that you can rely on to address and solve Data Science problems in the complexity of the real world (which is messy by definition).
+
+In this guide, you'll find a ton of different small tools, big libraries, and even entire frameworks, but be aware!
+
+As stated in the [Virgilio's Teaching Strategy](https://github.com/virgili0/Virgilio/blob/master/serving/paradiso/virgilio-teaching-strategy/virgilio-teaching-strategy.md) Guide, the best way to deal with the learning process is to leverage the "[Divide et Impera](https://en.wikipedia.org/wiki/Divide_and_rule)" principle, which states that every time you're tackling a problem (in this case, the _learning problem_) you should split it into smaller parts, easier to handle and understand. 
+
+# Solving the Learning Problem 
+
+The process of learning a tool and use it effectively **is equal to solving a problem**.
+
+We consider "solved" the problem when you can use the tool comfortably.
```

Comment:
This sentence does not read clear to me.

Reply:
slightly rewritten

Reply:
Did you mean to say
> We consider the problem "solved" when you can use the tool comfortably.